### PR TITLE
add Sign support

### PR DESCRIPTION
**Pull request by lucienwang1009**

### CODE REVIEW

File: tf2onnx/tfonnx.py [hunk 1]
```diff
@@ -866,6 +866,24 @@ def cast_op(ctx, node, name, args):
...
+    zero_name = utils.make_name("{}_zero".format(node.name))
+    ctx.make_const(zero_name, np.array(0, dtype=input_tensor_type))
+    greater_node = ctx.make_node("Greater", [node.input[0], zero_name])
+    less_node = ctx.make_node("Less", [node.input[0], zero_name])
+    cast_node_1 = ctx.make_node("Cast", [greater_node.output[0]], {"to": node_dtype})
+    cast_node_2 = ctx.make_node("Cast", [less_node.output[0]], {"to": node_dtype})
+    sub_node = ctx.make_node("Sub", [cast_node_1.output[0], cast_node_2.output[0]], outputs=[node.output[0]])
+    nodes.extend([greater_node, less_node, cast_node_1, cast_node_2, sub_node])
+    return nodes
+
```
Comment (pengwa) [inline tf2onnx/tfonnx.py]: two spaces between functions

File: tf2onnx/tfonnx.py [hunk 2]
```diff
@@ -866,6 +866,24 @@ def cast_op(ctx, node, name, args):
...
 
 
+def sign_op(ctx, node, name, args):
+    """Sign op."""
+    # T sign = Sign(T Input)
+    nodes = []
+    node_dtype = node.dtype
+    if node_dtype in [onnx_pb.TensorProto.COMPLEX64, onnx_pb.TensorProto.COMPLEX128]:
+        raise ValueError("dtype " + node.dtype + " is not supported in onnx for now")
+    input_tensor_type = utils.ONNX_TO_NUMPY_DTYPE[node_dtype]
```
Comment (pengwa) [inline tf2onnx/tfonnx.py]: utils.make_sure() to make sure node_dtype is not null

Comment (lucienwang1009) [reply tf2onnx/tfonnx.py]: sure, btw: why don't put this check into `dtype` method?

**Comment by lucienwang1009** [comment]
kick off a new CI build plea.

**Comment by pengwa** [comment]
restarted the build.

File: tests/test_backend.py [hunk 3]
```diff
@@ -858,6 +858,21 @@ def test_cast(self):
         _ = tf.identity(x_, name=_TFOUTPUT)
         self._run_test_case([_OUTPUT], {_INPUT: x_val})
 
+    @unittest.skipIf(OPSET < 9, "supported since opset 9")
```
Comment (pengwa) [inline tests/test_backend.py]: change to "int is supported since opset 9"?

Comment (pengwa) [reply tests/test_backend.py]: you can change it in next pr

**Pull request merged**